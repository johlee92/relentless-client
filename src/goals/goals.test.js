import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Goals from './goals';
import { BrowserRouter } from 'react-router-dom';

describe(`Goals component`, () => {
    const props = {
        className: 'test-class-name',
        viewGoals: 'annual',
        currentGoals: {
            weeklyGoals: [],
            monthlyGoals: [],
            annualGoals: ['test']
        }
    }

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter> 
                <Goals {...props}/>
            </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the Goals component from props', () => {
        const wrapper = shallow(<Goals {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})
