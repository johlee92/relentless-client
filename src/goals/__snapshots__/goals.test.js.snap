// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Goals component renders the Goals component from props 1`] = `
<div>
  <GoalCards
    goal="test"
    key="annual: undefined"
    viewGoals="annual"
  />
</div>
`;
